docs/source/tutorials/improving_ml_performance.ipynb

```diff
@@ -0,0 +1,3573 @@
+      "Best parameters: {'drop_label_issue': 0.6, 'drop_near_duplicate': 0.2, 'drop_outlier': 0.5}\n"
+     ]
+    }
+   ],
+   "source": [
+    "best_score = 0\n",
+    "best_params = None\n",
+    "\n",
+    "for drop_label_issue, drop_near_duplicate, drop_outlier in param_combinations:\n",
+    "    # Preprocess the data for the current combination of parameters\n",
+    "    train_features_preprocessed, train_labels_preprocessed = preprocess_data(\n",
+    "        train_features_v2, train_labels_v2, label_issues, near_duplicates, outliers,\n",
+    "        drop_label_issue, drop_near_duplicate, drop_outlier)\n",
+    "    \n",
+    "    # Train and evaluate the model\n",
+    "    model = XGBClassifier(tree_method=\"hist\", enable_categorical=True, random_state=SEED)\n",
+    "    model.fit(train_features_preprocessed, train_labels_preprocessed)\n",
+    "    predictions = model.predict(test_features)\n",
+    "    accuracy = accuracy_score(test_labels, predictions)\n",
+    "    \n",
+    "    # Update the best score and parameters if the current model is better\n",
+    "    if accuracy > best_score:\n",
+    "        best_score = accuracy\n",
+    "        best_params = {'drop_label_issue': drop_label_issue, 'drop_near_duplicate': drop_near_duplicate, 'drop_outlier': drop_outlier}\n",
+    "\n",
+    "# Print the best parameters and score\n",
+    "print(f\"Best parameters: {best_params}\")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 47,
+   "id": "3f572acf-31c3-4874-9100-451796e35b06",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Accuracy of model fit to clean training data based on the optimal combinations of hyperparameters to clean our data, measured on clean test data: 80.2%\n"
+     ]
+    }
+   ],
+   "source": [
+    "print(\n",
+    "    f\"Accuracy of model fit to clean training data based on the optimal combinations of hyperparameters to clean our data, measured on clean test data: {round(best_score*100,1)}%\"\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "cc306eff-f3b7-4098-9f7e-3d17d1d0016a",
+   "metadata": {},
+   "source": [
+    "`cleanlab` was able to successfully help us improve ML performance in this tutorial! We saw how `cleanlab` helped us find and manually fix different data issue types in our test data to clean it. \n",
+    "\n",
+    "Then we fit a model on our noisy training data but evaluated it on our clean test data. We cleaned our training data by first dropping all rows for each issue type and fit a new model on our clean training data and evaluated it on our clean test data (and saw improvement already in model accuracy)! \n",
+    "\n",
+    "We then were able to further improve model accuracy by optimizing for the exact amount of each issue type to drop from our data using hyperparameter optimization. \n",
+    "\n",
+    "To reiterate, here are the 2 main takeaways:\n",
```

Comment:
These aren't my main takeaway from reading this! My main takeaway is that we should use cleanlab to optimize cleaning of a dataset. 

Belaboring my point here but, instead of saying "DONT DO X" can you instead find a way to make it positive or at the very least explain _why_ this is a bad idea? 

I'm super happy to help - some quick post like "The folly of algorithmically cleaning test data", although it's a kind of confusing message because I think folks will think part of the point of cleanlab is to automatically clean data.

Reply:
@nelsonauner Appreciate the detailed review!  
I think you might be missing some context, so will provide just in case you want to revise any suggestions.

Some users want to use cleanlab to improve their ML model by improving their data. However the proper way to do this is highly subtle, hence we made this complex tutorial about it. The goal of the tutorial isn't to encourage users to use cleanlab to improve their ML model, it's for users who already want to do that to do it right.

We see some of these users curate (eg. fix label issues in) their training set, then train ML model, and eval on test data. But then they see no improvement in test-set accuracy. Why?  Because they have actually introduced distribution-shift by altering their training data.  If the test data still has issues, they must also be fixed for a faithful model evaluation.

Other users realize this and curate their test data too. But some of these users auto-fix their test data, which is dangerous!  Cleanlab is imperfect (based on ML which is inevitably imperfect).  Test data should NOT be blindly auto-fixed, that risks making model evaluation wrong. 

Instead the recommended approach is the multi-step workflow outlined at the top of the tutorial, where less algorithmic/automated correction is applied to test data than to training data (focus manual effort/reviewing budget on curating test data rather than train)

Reply:
Thanks for the additional context. 

I feel that your explanation in the above comment is incredibly useful and deserves to be _somewhere_  in the tutorial, ideally next to the text where we tell them not to do it. But it's not a hill I'd die on!

Reply:
added at end of intro